modules/social_features/social_core/src/Controller/SocialCoreController.php

```diff
@@ -167,65 +200,90 @@ public function updateSelection($view_id, $display_id, Request $request) {
+      ? $invite
+      // If there are no invites found or no module enabled lets redirect user
+      // to their stream.
+      : 'social_user.user_home';
 
-    // If there are no invites found or no module enabled
-    // lets redirect user to their stream.
-    return $this->redirect('social_user.user_home', [
-      'user' => $this->currentUser()->id(),
-    ]);
+    return $this->redirect($route_name, ['user' => $this->currentUser()->id()]);
   }
 
   /**
-   * The _title_callback for the node.add route.
-   *
-   * @param \Drupal\node\NodeTypeInterface $node_type
-   *   The current node.
-   *
-   * @return string
-   *   The page title.
+   * The _title_callback for the entity creation route.
    */
-  public function addPageTitle(NodeTypeInterface $node_type) {
-    // The node_types that have a different article than a.
-    $node_types = [
-      'event' => 'an',
-    ];
+  public function addPageTitle(): ?TranslatableMarkup {
+    $titles = $this->moduleHandler()->invokeAll('social_core_title');
+    $this->moduleHandler()->alter('social_core_title', $titles);
+
+    if (!empty($titles['node'])) {
+      if (!isset($titles['node']['bundles'])) {
+        $titles['node']['bundles'] = [];
+      }
 
-    // Make sure extensions can change this as well.
-    \Drupal::moduleHandler()->alter('social_node_title_prefix_articles', $node_types);
+      $this->moduleHandler()->alterDeprecated(
+        'Deprecated in social:11.2.0 and is removed from social:12.0.0. Use hook_social_core_title_alter instead. See https://www.drupal.org/node/3285045',
+        'social_node_title_prefix_articles',
+        $titles['node']['bundles'],
+      );
+    }
+
+    if (($route_name = $this->routeMatch->getRouteName()) === NULL) {
+      return NULL;
+    }
+
+    $route_name = explode('.', $route_name);
+    $entity_type_id = $route_name[count($route_name) - 2];
+
+    if (isset($titles[$entity_type_id]['callback'])) {
+      $entity_type = $titles[$entity_type_id]['callback']();
+    }
+    else {
+      $definition = $this->entityTypeManager()->getDefinition($entity_type_id);
+
+      if (
+        $definition !== NULL &&
+        ($bundle_entity_type = $definition->getBundleEntityType()) !== NULL
+      ) {
+        $entity_type = $this->routeMatch->getParameter($bundle_entity_type);
+      }
+    }
 
-    if ($node_type !== NULL && array_key_exists($node_type->id(), $node_types)) {
+    if (
+      isset($entity_type) &&
+      $entity_type instanceof EntityInterface &&
+      ($label = $entity_type->label()) !== NULL
+    ) {
       return $this->t('Create @article @name', [
-        '@article' => $node_types[$node_type->id()],
-        '@name' => $node_type->label(),
+        '@article' => $titles[$entity_type_id]['bundles'][$entity_type->id()] ?? 'a',
+        '@name' => mb_strtolower($label),
       ]);
     }
 
-    return $this->t('Create a @name', ['@name' => $node_type->label()]);
+    return NULL;
```

Comment:
Previously this hook was retuning only string, and now it is returning string or null, this might break someones code that is not considering this new return type, please keep it the same return that it was, if possible.

Reply:
I see it was fixed and now returns string type as it was before.

Reply:
You're right again.